### PR TITLE
refactor(titus): Allow serde of operation descriptions

### PR DESCRIPTION
**Pull request by robzienert**
This was about the smallest I could make the PR without jumping through a ton of hoops.

The original pass of events/sagas stored everything in memory and didn't uncover an issue where clouddriver descriptions are not deserialization-friendly. This led me down a path of some fairly aggressive refactoring.

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/config/CloudDriverConfig.groovy [hunk 1]
```diff
@@ -127,6 +128,7 @@ class CloudDriverConfig {
         jacksonObjectMapperBuilder.serializationInclusion(JsonInclude.Include.NON_NULL)
         jacksonObjectMapperBuilder.failOnEmptyBeans(false)
         jacksonObjectMapperBuilder.failOnUnknownProperties(false)
+        jacksonObjectMapperBuilder.modules(new Jdk8Module(), new JavaTimeModule(), new KotlinModule())
```
Comment (robzienert) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/config/CloudDriverConfig.groovy]: Clouddriver only has one `ObjectMapper`, which is configured in core. Unfortunately, this mapper builder way of customizing things does not expose the `findAndRegisterModules` capability, for whatever reason, so we have to be explicit.

File: clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/SubmitTitusJob.java [hunk 2]
```diff
@@ -98,7 +90,7 @@ public Result apply(@NotNull SubmitTitusJobCommand command, @NotNull Saga saga)
         retrySupport.retry(
             () -> {
               try {
-                return titusClient.submitJob(submitJobRequest);
+                return titusClient.submitJob(submitJobRequest.withJobName(nextServerGroupName[0]));
```
Comment (robzienert) [inline clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/SubmitTitusJob.java]: This class is so upsetting, but there was already enough change in this PR. I want to come back and make this class not super gross.

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt [hunk 3]
```diff
@@ -49,8 +51,30 @@ interface EventRepository {
...
+   * @return A list of matching aggregates
    */
-  fun listAggregates(aggregateType: String?): List<Aggregate>
+  fun listAggregates(criteria: ListAggregatesCriteria): ListAggregatesResponse
+
+  /**
+   * @param aggregateType The type of [Aggregate] to return. If unset, all types will be returned.
+   * @param token The page token to paginate from. It will return the first results
+   * @param perPage The number of [Aggregate]s to return in each response
+   */
```
Comment (srekapalli) [inline clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt]: Confused by the 'name' of this class.  This represents AggregateCriteria or you meant list of aggregate criteria objects ?

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt [hunk 4]
```diff
@@ -49,8 +51,30 @@ interface EventRepository {
...
+  class ListAggregatesCriteria(
+    val aggregateType: String? = null,
+    val token: String? = null,
+
+    @Positive @Max(1000)
+    val perPage: Int = 100
+  )
+
+  /**
```
Comment (srekapalli) [inline clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt]: Is this representing Response for a Aggregate request ?

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/ComposedSpinnakerEvent.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ */
+package com.netflix.spinnaker.clouddriver.event
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+
+/**
+ * Marks a [SpinnakerEvent] as being constructed of multiple [SpinnakerEvent]s.
+ *
+ * This interface is necessary to correctly hydrate [EventMetadata] on [SpinnakerEvent] before persisting.
+ */
```
Comment (srekapalli) [inline clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/ComposedSpinnakerEvent.kt]: Is this a 'Composite' or 'Composed' ?

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt (hunk 3 above)
Comment (robzienert) [reply clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt]: It's the criteria for the `listAggregates` method. The response of that method will conform to the criteria defined by this object (if any).

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt (hunk 4 above)
Comment (robzienert) [reply clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt]: It's representing the response from the `listAggregates` method. Not to be confused with an HTTP response or something else. Would it make more sense if it were renamed ot `ListAggregatesResult`?

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/ComposedSpinnakerEvent.kt (hunk 5 above)
Comment (robzienert) [reply clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/ComposedSpinnakerEvent.kt]: Will rename to `CompositeSpinnakerEvent`, much bester.

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt (hunk 4 above)
Comment (srekapalli) [reply clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt]: +1 for ListAggregatesResult

**Pull request merged**